src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs

```diff
@@ -309,17 +398,17 @@ private MessageHandlerOptions DetermineMessageHandlerOptions(AzureServiceBusMess
             return messageHandlerOptions;
         }
 
-        private async Task ReAuthenticateMessageReceiverAsync(Exception exception)
+        /// <summary>
+        /// Restart core functionality of the message pump.
+        /// </summary>
+        public async Task RestartAsync()
         {
-            if (exception is UnauthorizedException)
```

Comment:
I don't know if we should remove this check maybe? While this is a great feature, we should not enforce people to use it, no?

Reply:
Aha, hmm, maybe make this a nullable integer so we only use the feature if the user set this value?

Reply:
Not sure if I'm following because this is related to UnauthorizedException?

Reply:
This is changed to a system where we count the amount of `UnauthorizedException`'s that's being thrown. A new configuration option is added that the user can configure. Before this PR we always restarted when we received a new `UnauthorizedException` but now we have to wait till the "Secret New Version Created" event is raised (when configured); so the user should also indicate how long they should wait for such event. That's why a new configuration option is added (`integer`). (also explained in the description of this PR).

My question here was, maybe we still check for `UnauthorizedException` types and count the amount of exceptions being thrown, but only when there's such a value present. AKA making it nullable.

Reply:
Although, I think I'm not such a fan of removing the feature like it is now (by default), since this way we can use both features at the same time by default (default 5 `UnauthorizedException`'s can be throwed before the pump restarts).

Reply:
Well for me it's important that we don't force people to rely on Azure Event Grid, but just give them the option. So I would support both

Reply:
Not sure how Azure Event Grid comes into play here, as these Key Vault events are put on a Service Bus Topic. To get back to your original question, if we support them both, then we should indeed leave it the way it is. But can maybe be discussed offline.

Reply:
Yes but Event Grid puts them on the topic, but that was indeed what I was referring to.

Reply:
Feel free to resolve if both are supported now

Reply:
The key vault event is optional, and can be combined with the already restart functionality, as explained in the meeting we had. I'll resolve this. You can look it over in mor detail once you have the time and we can do another checkup if needed.